src/system/user.cr

```diff
@@ -103,14 +104,14 @@ struct System::User
   # ```
   # System::User.get("root").uid # => 0
   # ```
-  getter uid : UInt32
+  getter uid : UID
 
   # Returns the users group ID.
   #
   # ```
   # System::User.get("root").gid # => 0
   # ```
-  getter gid : UInt32
+  getter gid : Group::GID
```

Comment:
We typically don't want to change the types of numbers based on the platform - unlike C we don't auto-cast between numbers so picking a suitable value and sticking to it is best.

I suggest keeping this a `UInt32` and casting.